src/leetCodeExplorer.ts

```diff
@@ -64,57 +73,100 @@ export class LeetCodeTreeDataProvider implements vscode.TreeDataProvider<LeetCod
-                        locked: false,
-                        state: ProblemState.Unknown,
-                        id: "notSignIn",
-                        name: "Sign in to LeetCode",
-                        difficulty: "",
-                        passRate: "",
-                    },
-                    false,
-                ),
+                new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                    id: "notSignIn",
+                    name: "Sign in to LeetCode",
+                }), false),
             ];
         }
-        if (!element) {
+        if (!element) { // Root view
             return new Promise(async (resolve: (res: LeetCodeNode[]) => void): Promise<void> => {
                 await this.getProblemData();
-                resolve(this.composeDifficultyNodes());
+                resolve([
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Difficulty",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Tag",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Company",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Favorite",
+                    }), false),
+                ]);
             });
         } else {
-            return element.isProblem ? [] : this.composeProblemNodes(element.name);
+            switch (element.name) { // First-level
+                case "Difficulty":
+                case "Tag":
+                case "Company":
+                    return this.composeCategoryNodes(element);
+                case "Favorite":
+                    return this.treeData.Favorite.map((p: list.IProblem) => new LeetCodeNode(p));
+                default: // Second and lower levels
+                    return element.isProblem ? [] : this.composeProblemNodes(element);
+            }
         }
     }
 
     private async getProblemData(): Promise<void> {
-        const allProblems: list.IProblem[] = await list.listProblems();
-        this.treeData.clear();
-        for (const problem of allProblems) {
-            const problems: list.IProblem[] | undefined = this.treeData.get(problem.difficulty);
-            if (problems) {
-                problems.push(problem);
-            } else {
-                this.treeData.set(problem.difficulty, [problem]);
+        this.treeData = {
+            Difficulty: new Map(),
+            Tag: new Map(),
+            Company: new Map(),
+            Favorite: [],
+        };
+        for (const problem of await list.listProblems()) {
```

Comment:
I'm thinking that we can filter the accepted problems in the method `list.listProblems()`. If `hideSolved === true` then only return unaccepted problems.

By doing this, we decouple the `this.leetCodeConfig` with the explorer

Reply:
In my opinion, `Favorite` group should show all starred problems no matter whether it's accepted or not. Filtering problems in `list.listProblems()` seems too early for this requirement.

Reply:
Hmm, ok, just keep it there first.